### PR TITLE
[LG-178] OCSP support for piv/cac certs

### PR DESCRIPTION
**Pull request by jgsmith-usds**
**Why**:
CRLs are a nice fall-back, but OCSP can give more timely
information on revocation status.

**How**:
We use the OpenSSL OCSP support to construct and interpret
OCSP requests/responses. We cache any positive revocation that
we receive.

### CODE REVIEW

File: app/services/certificate_store.rb [hunk 1]
```diff
@@ -86,21 +89,22 @@ def all_certificates_valid?
...
-  class << self
-    def trusted_ca_root_identifiers
-      @trusted_ca_root_identifiers ||= begin
-        raw = Figaro.env.trusted_ca_root_identifiers || ''
-        raw.split(',').map(&:strip) - ['']
-      end
-    end
+  def self.trusted_ca_root_identifiers
+    @trusted_ca_root_identifiers ||=
+      (Figaro.env.trusted_ca_root_identifiers || '').split(',').map(&:strip) - ['']
```
[the remark this answers is not in the document]
Comment (jgsmith-usds) [reply app/services/certificate_store.rb]: It should already be there. This isn't adding a new configuration key.

**Comment by jgsmith-usds** [comment]
Will do on the README. We'll need proxy rules for the OCSP servers. I'll send a list of those I know about.

**Comment by konklone** [comment]
cc @lachellel for thoughts/visibility

**Comment by lachellel** [comment]
@konklone I don't see the latest list of issuing CAs accepted.  If you point me to that list, we can provide the **end entity** OCSP URIs.  Only the end entity are required to have OCSP URIs.  And in some cases, two different end entity certs signed by the _same_ issuing CA will have different OCSP URIs.  [I can identify two services where this is fact.]

Also, if you only want the issuing CAs that issue PIV/CAC ->here is the list:  https://federalist-proxy.app.cloud.gov/preview/gsa/fpki-guides/97-TSMS-CLEAN/tsmseligiblecacerts/

Other issuing CAs may sign certs for other purposes, including PIV-Interoperable, dig sig (not used for client auth), or solely device certs.

**Comment by jgsmith-usds** [comment]
The page at https://federalist-proxy.app.cloud.gov/preview/gsa/fpki-guides/97-TSMS-CLEAN/tsmseligiblecacerts/ has a list of certs but doesn't provide the certs. Is there place where we can find the certs?

**Comment by lachellel** [comment]
@jgsmith-usds it's in a zip linked from the other page.  I should have put this link instead (sorry!): 

https://federalist-proxy.app.cloud.gov/preview/gsa/fpki-guides/97-TSMS-CLEAN/tools/trust_store_management/#download-script-package

There are 45 **issuing** CA public certificates in there.

**Comment by jgsmith-usds** [comment]
Thanks! I'll take a look at the bundle.

**Pull request merged**